### PR TITLE
grant_sprint4/hiddencard

### PR DESCRIPTION
**Pull request by grant-warkentin**
1.) added functionality to the showHand() function so it will hide the second card of the first two cards when directed to.
2.) I also made sure the sum of the hand would only be of the card showing by adding some code to handSum()
3.) fixed a bug that didn't allow you to bet your entire available amount
4.) added code to the blackjack() function so it doesn't allow betting $0. Something must be wagered.
5.) Other minor code improvements to the blackjack() function.

### CODE REVIEW

File: source/CardGames.py [hunk 1]
```diff
@@ -130,7 +131,7 @@ def makeBet(self, amount: int):
       print(f"{self.name} has {self.money} dollars. This is not enough to make this bet.")
       return False
     self.money -= amount
-    return self.money
+    return self.money, True
```
Comment (TommyMedina) [inline source/CardGames.py:134]: why did you add ", True" ? what purpose did it serve?

File: source/CardGames.py [hunk 2]
```diff
@@ -279,133 +290,141 @@ def winnnerCheck(self, house, player):
...
-    else:
-      house.clearHand()
-      player_name.clearHand()
+      if deck.size <=52 and early_shuffle:
+        dealer.resetDeck()
 
-    if deck.size <=52 and early_shuffle:
-      dealer.resetDeck()
\ No newline at end of file
+#blackjack()
```
Comment (TommyMedina) [inline source/CardGames.py]: this need to not be commented out for your code to run. Depending on why you commented that out. this might be a more robust solition
 if __name__ == '__main__':
 blackjack()

no matter what blackjack needs to be called for this code to be merged

Comment (grant-warkentin) [reply source/CardGames.py]: When I was running my test it also ran blackjack in the source file. I just commented it out as a quick solution so that wouldn't happen. I'll make sure it runs, thanks!

File: source/CardGames.py (hunk 1 above)
Comment (grant-warkentin) [reply source/CardGames.py:134]: By default, if return True isn't specified return None will be given and interpreted by python as a false value. Therefore, I needed a return True, without it this while loop in the blackjack() function was triggering.  This was part of the error that disallowed a max bet.  

while player_name.makeBet(bet) == False:
     bet = int(input("\nHow much would you like to bet? "))

Comment (TommyMedina) [reply source/CardGames.py:134]: Thank you, that helps.

**Review by Alex-Church — APPROVED** [review]
Looks good from what I can see. Did you update the code with the function call?

**Review by TommyMedina — APPROVED** [review]

File: source/CardGames.py [hunk 3]
```diff
@@ -139,10 +140,14 @@ def addCard(self, card: Card, isKnown: bool = True):
...
       self.knownCards.append(False)
 
-  def handSum(self):
+  def handSum(self, hideSum: bool = False):
     """Sums together the hand and adjusts the sum depending on the best value of an ace (11 or 1)"""
     handsum = 0
     aces = 0
+    if hideSum: # hides the value of the first card when calculating the sum
+      card = self.hand[0]
+      handsum += card.value
```
Comment (php-121) [inline source/CardGames.py:149]: can combine 2 lines into: handsum += self.hand[0].value

File: source/CardGames.py [hunk 4]
```diff
@@ -156,18 +161,24 @@ def setHand(self, cards: "list[Card]", isKnown: bool = False):
     self.hand = cards
     self.knownCards = [isKnown for _ in self.hand]
 
-  def showHand(self, printShort: bool = False):
+  def showHand(self, printShort: bool = False, showBack: bool = False):
     for idx in range(6):
       for i, card in enumerate(self.hand):
```
Comment (php-121) [inline source/CardGames.py:166]: Having idx and i in the same loop is confusing to me since i stands for index. Usually I use i and j

File: source/CardGames.py [hunk 5]
```diff
@@ -279,133 +290,141 @@ def winnnerCheck(self, house, player):
...
-    early_shuffle = True
+def blackjack():
+  """Initializations"""
+
+  player_name = input("Username: ")
+
+  partition = player_name.find(";")
+  config=""
+  if partition > 0:
+    config = player_name[partition:]
```
Comment (php-121) [inline source/CardGames.py]: can combine into 1 line:
config, player_name = player_name[partition:], player_name[0:partition]

**Review by php-121 — COMMENTED** [review]
just some comments about the format, otherwise LGTM
